packages/dds/tree/src/test/time/tableTree.spec.ts

```diff
@@ -0,0 +1,646 @@
+							const column = new Column({ id: `column-${uuid()}` });
+							const row = new Row({ id: `row-${uuid()}`, cells: {} });
+							table.insertColumn({ index: Math.floor(table.columns.length / 2), column });
+							table.insertRow({ index: Math.floor(table.rows.length / 2), row });
+						}
+					},
+					stackOperation: (tree: TableTreeDefinition) => {
+						const { undoStack } = tree;
+						// assert.equal(stack.undoStackLength, count);
+						for (let i = 0; i < 2 * count; i++) {
+							undoStack.pop()?.revert();
+						}
+					},
+					maxBenchmarkDurationSeconds,
+				});
+
+				// Test the execute time of the SharedTree for redoing an insert row and a column in the middle for a given number of times.
+				runUndoRedoBenchmark({
+					title: `Redo insert the middle column and row ${count} times`,
+					tableSize: treeSize,
+					cellValue,
+					setupOperation: (tree: TableTreeDefinition) => {
+						const { Column, Row, table, undoStack } = tree;
+						for (let i = 0; i < count; i++) {
+							const column = new Column({ id: `column-${uuid()}` });
+							const row = new Row({ id: `row-${uuid()}`, cells: {} });
+							table.insertColumn({ index: Math.floor(table.columns.length / 2), column });
+							table.insertRow({ index: Math.floor(table.rows.length / 2), row });
+						}
+						for (let i = 0; i < 2 * count; i++) {
+							undoStack.pop()?.revert();
+						}
+					},
+					stackOperation: (tree: TableTreeDefinition) => {
+						const { redoStack } = tree;
+						// assert.equal(stack.undoStackLength, count);
+						for (let i = 0; i < 2 * count; i++) {
+							redoStack.pop()?.revert();
+						}
+					},
+					maxBenchmarkDurationSeconds,
+				});
+			});
+		}
+
+		// Set/Remove-related tests that are limited by treeSize
+		for (const count of validRemoveCounts) {
+			describe(`Column Removal`, () => {
+				// Test the execute time of the SharedTree for removing a column in the middle for a given number of times.
+				runBenchmark({
+					title: `Remove a column in the middle ${count} times`,
+					tableSize: treeSize,
+					cellValue,
+					operation: (tree: TableTreeDefinition) => {
+						const { table } = tree;
+						for (let i = 0; i < count; i++) {
+							table.removeColumn(`column-${i}`);
+						}
+					},
+					maxBenchmarkDurationSeconds,
+				});
+
+				// Test the execute time of undoing remove a column in the middle for a given number of times.
+				runUndoRedoBenchmark({
+					title: `Undo remove the middle column ${count} times`,
+					tableSize: treeSize,
+					cellValue,
+					setupOperation: (tree: TableTreeDefinition) => {
+						const { table } = tree;
+						for (let i = 0; i < count; i++) {
+							table.removeColumn(`column-${i}`);
+						}
+					},
+					stackOperation: (tree: TableTreeDefinition) => {
+						const { undoStack } = tree;
+						// assert.equal(stack.undoStackLength, count);
+						for (let i = 0; i < count; i++) {
+							undoStack.pop()?.revert();
```

Comment:
These tests are structured to ensure we know how many operations we have to undo. It would be a problem if `pop()` didn't give us back a revertible (and would indicate a bug in either the tests or the underlying code). In normal circumstances, I would recommend adding an assert to verify that `undoStack.pop` didn't return undefined. But since we're doing performance measurements here, I think it may be better to just do the following instead:
```suggestion
							undoStack.pop()!.revert();
```

You may need to disable a linter rule for this, for which we can document the reasoning.

If `pop` returns undefined, then calling `revert` will throw an exception that will fail the test, which we want.

Reply:
Updated